<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>842</width>
    <height>715</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>oct_reverse_tool v01 by wangdonghao </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>选择缩放的对象：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="scale_obj_lineEdit">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>30</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="scale_obj_pushButton">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>30</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>《《</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>选择反转的对象：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="reverse_lineEdit">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>180</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="reverse_pushButton">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>180</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>《《</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>70</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>缩放属性：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>220</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>反转属性：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="doit_pushButton">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>550</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(90, 168, 78);</string>
    </property>
    <property name="text">
     <string>执行！</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>60</y>
      <width>61</width>
      <height>62</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QRadioButton" name="scaleX">
       <property name="text">
        <string>scaleX</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="scaleY">
       <property name="text">
        <string>scaleY</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="scaleZ">
       <property name="text">
        <string>scaleZ</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>230</y>
      <width>85</width>
      <height>62</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QRadioButton" name="translateX">
       <property name="text">
        <string>translateX</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="translateY">
       <property name="text">
        <string>translateY</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="translateZ">
       <property name="text">
        <string>translateZ</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>310</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>需要排除的节点：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="except_pushButton">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>310</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>《</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>490</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>L/R当前：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>490</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>L/R非当前：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="l_format_lineEdit">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>490</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>_R</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="r_format_lineEdit">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>490</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>_L</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidget">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>310</y>
      <width>241</width>
      <height>161</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>250</number>
    </attribute>
    <column>
     <property name="text">
      <string>remove_node</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="except_pushButton_2">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>350</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="edit_cfg_pushButton">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>70</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>    编辑属性单》</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="rotate_cons">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>280</y>
      <width>171</width>
      <height>181</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>250</number>
    </attribute>
    <column>
     <property name="text">
      <string>remove_node</string>
     </property>
    </column>
   </widget>
   <widget class="QLineEdit" name="rotate_attr">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>480</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="input_rotate_attr">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>480</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>《通道栏获取</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>240</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>反转Rotate：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>280</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>控制器：</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>0</y>
      <width>20</width>
      <height>661</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="reverse_rotation_doit">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>550</y>
      <width>231</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(212, 132, 108);
</string>
    </property>
    <property name="text">
     <string>执行！</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>480</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>属性：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>反转Translate：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clear_rotate_cons">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>280</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>clear</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="is_matrix_checkBox">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>260</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>是否是matrix控制形变</string>
    </property>
   </widget>
   <widget class="QPushButton" name="restore_shape_pushButton">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>600</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>强制恢复形状！</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="undo_count_lineEdit">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>10</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>10</string>
    </property>
   </widget>
   <widget class="QPushButton" name="undo_pushButton">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>5</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>高效率Undo</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>842</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
